python/pyspark/sql/tests/test_utils.py

```diff
@@ -45,9 +45,14 @@
     IntegerType,
     BooleanType,
 )
-from pyspark.testing.sqlutils import have_pandas
+from pyspark.testing.sqlutils import (
+    have_pandas,
+    have_pyarrow,
+    pyarrow_requirement_message,
+)
 
 
+@unittest.skipIf(not have_pyarrow, pyarrow_requirement_message)
 class UtilsTestsMixin:
```

Comment:
why does this require pyarrow and pandas? There's something wrong in the code base now.

Reply:
Because the method `assertDataFrameEqual` is used inside, it requires `pyarrow`.

Reply:
Hmm,, I don't think `assertDataFrameEqual` uses `pyarrow`?? It uses `pandas`, but it's not required package because we just skip using pandas when it's not installed. Could check [comment](https://github.com/apache/spark/pull/44778#discussion_r1464195098) above?

Reply:
Okay